zerver/tests/test_user_groups.py

```diff
@@ -152,67 +156,85 @@ def test_recursive_queries_for_user_groups(self) -> None:
         self.assertCountEqual(
-            list(get_recursive_subgroups(admins_group)), [owners_group, admins_group]
+            list(get_recursive_subgroups(owners_group)), [owners_group.usergroup_ptr]
+        )
+        self.assertCountEqual(
+            list(get_recursive_subgroups(admins_group)),
+            [owners_group.usergroup_ptr, admins_group.usergroup_ptr],
         )
         self.assertCountEqual(
             list(get_recursive_subgroups(moderators_group)),
-            [owners_group, admins_group, moderators_group],
+            [
+                owners_group.usergroup_ptr,
+                admins_group.usergroup_ptr,
+                moderators_group.usergroup_ptr,
+            ],
         )
         self.assertCountEqual(
             list(get_recursive_subgroups(full_members_group)),
-            [owners_group, admins_group, moderators_group, full_members_group],
+            [
+                owners_group.usergroup_ptr,
+                admins_group.usergroup_ptr,
+                moderators_group.usergroup_ptr,
+                full_members_group.usergroup_ptr,
+            ],
         )
         self.assertCountEqual(
             list(get_recursive_subgroups(members_group)),
-            [owners_group, admins_group, moderators_group, full_members_group, members_group],
+            [
+                owners_group.usergroup_ptr,
+                admins_group.usergroup_ptr,
+                moderators_group.usergroup_ptr,
+                full_members_group.usergroup_ptr,
+                members_group.usergroup_ptr,
+            ],
         )
         self.assertCountEqual(
             list(get_recursive_subgroups(everyone_group)),
             [
-                owners_group,
-                admins_group,
-                moderators_group,
-                full_members_group,
-                members_group,
-                everyone_group,
+                owners_group.usergroup_ptr,
+                admins_group.usergroup_ptr,
+                moderators_group.usergroup_ptr,
+                full_members_group.usergroup_ptr,
+                members_group.usergroup_ptr,
+                everyone_group.usergroup_ptr,
             ],
         )
         self.assertCountEqual(
             list(get_recursive_subgroups(everyone_on_internet_group)),
             [
-                owners_group,
-                admins_group,
-                moderators_group,
-                full_members_group,
-                members_group,
-                everyone_group,
-                everyone_on_internet_group,
+                owners_group.usergroup_ptr,
+                admins_group.usergroup_ptr,
+                moderators_group.usergroup_ptr,
+                full_members_group.usergroup_ptr,
+                members_group.usergroup_ptr,
+                everyone_group.usergroup_ptr,
+                everyone_on_internet_group.usergroup_ptr,
```

Comment:
Hmm, could these changes be avoided? I think maybe at least here we could have the test be verifying `get_recursive_strict_subgroups`.

Reply:
This will happen for functions that return `UserGroup` as we query the `NamedUserGroup` object in tests. But yes, updated these tests to use `get_recursive_strict_subgroups`.